internal/providers/docker/compose/project.go

```diff
@@ -20,12 +21,24 @@ import (
 func Parse(r io.Reader) (*Compose, error) {
 	dec := yaml.NewDecoder(r,
 		yaml.DisallowDuplicateKey(),
-		yaml.DisallowUnknownField(), // TODO: Handle this more gracefully.
 	)
 	var comp Compose
 	if err := dec.Decode(&comp); err != nil {
 		return nil, err
 	}
+
+	// Validate.
+	for key := range comp.Raw {
+		switch key {
+		case "version", "services", "networks", "volumes", "configs", "secrets":
+			// Ok.
+		default:
+			if !strings.HasPrefix(key, "x-") {
```

Comment:
What's the `x-` prefix?

Reply:
https://github.com/compose-spec/compose-spec/blob/master/spec.md#extension